### PR TITLE
ddl: pause subtask when exec ddl err

### LINKED ISSUE #984: handle schema tracker newer than downstream TiDB
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

update schema tracker in https://github.com/pingcap/dm/pull/982 to enjoy more compatibility, some test failed, such as alter FK without schema.

2. What did you expect to see?

`query-status` shows nothing

3. What did you see instead?

`query-status` shows that error because downstream TiDB doesn't support

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

       after #982

    - Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

       older than `20200901032733-f82e5320ad75`

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #984 

### What is changed and how it works?
- report error when execute ddl fail in dowsteam TiDB
- omit error in syncDDL caller to avoid duplicate error message

### Check List 

Tests 

 - Integration test

### CODE REVIEW

File: syncer/syncer.go [hunk 1]
```diff
@@ -931,11 +931,18 @@ func (s *Syncer) syncDDL(tctx *tcontext.Context, queueBucket string, db *DBConn,
 			}
 		}
 		if err != nil {
+			s.jobWg.Done()
```
Comment (lance6716) [inline syncer/syncer.go:934]: not sure `s.jobWg.Done()` should be moved later than `s.execError.Set(err)`, to make sure caller will see the error after `s.jobWg.Wait()`

File: syncer/syncer.go [hunk 2]
```diff
@@ -931,11 +931,18 @@ func (s *Syncer) syncDDL(tctx *tcontext.Context, queueBucket string, db *DBConn,
...
+			if !utils.IsContextCanceledError(err) {
+				err = s.handleEventError(err, &sqlJob.startLocation, &sqlJob.currentLocation)
+				s.runFatalChan <- unit.NewProcessError(err)
+			}
 			s.appendExecErrors(&ExecErrorContext{
 				err:      err,
 				location: sqlJob.currentLocation.Clone(),
 				jobs:     fmt.Sprintf("%v", sqlJob.ddls),
 			})
+			continue
```
Comment (lance6716) [inline syncer/syncer.go:945]: forward: "can we just rollback checkpoint as we did before?" https://github.com/pingcap/dm/pull/1032#discussion_r489988212

I think that's no need, we could use caller's rollback

**Review by lance6716 — COMMENTED** [review]
rest LGTM

File: syncer/syncer.go (hunk 1 above)
Comment (GMHDBJD) [reply syncer/syncer.go]: I move all of it. Actually we now no need `s.execError.get()` in caller. If syncDDL error, subtask will pause and rollback checkpoint in syncer.Process

File: syncer/syncer.go (hunk 2 above)
Comment (GMHDBJD) [reply syncer/syncer.go:945]: So no need to change `addjob` in #1032, right?

**Comment by lance6716** [comment]
/run-all-tests

```
[2020-09-17T10:52:23.616Z] FAIL: server_test.go:1293: testMaster.TestOfflineMember
[2020-09-17T10:52:23.616Z] 
[2020-09-17T10:52:23.616Z] server_test.go:1357:
[2020-09-17T10:52:23.616Z]     c.Assert(err, check.IsNil)
[2020-09-17T10:52:23.616Z] ... value *status.statusError = &status.statusError{Code:2, Message:"[code=38043:class=dm-master:scope=internal:level=high], Message: master is not leader, and can't forward request to leader", Details:[]*any.Any(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0} ("rpc error: code = Unknown desc = [code=38043:class=dm-master:scope=internal:level=high], Message: master is not leader, and can't forward request to leader")
```

**Comment by lance6716** [comment]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**